rfc/2022/2022-07-20 - Update Build Directory Structure of vets-website.md

```diff
@@ -0,0 +1,61 @@
+
+Separating bundles and assets of isolated application builds from bundles and assets of full builds can prevent potentially breaking isolated applications in production.
+
+## Motivation
+
+Storing the bundles and assets of both build types separately would resolve issues that can occur from syncing isolated application bundles with global bundles and assets generated from previous deployments.
+
+## Design
+
+### Updating the structure of the `build` directory
+
+Restructuring the `build/generated` directory involves updating the `vets-website` build to not store bundles of isolated applications in the root of the directory. Doing so means that the root directory will only contain bundles of applications that aren't isolated, along with the global bundles and assets generated from full builds, or in this case, builds of non-isolated applications. Bundles and assets from isolated application builds will be stored in their own subdirectory within the `build/generated` directory, e.g. `build/generated/check-in`, `build/generated/vaos`, etc. Doing this will ensure two things:
+
+1. Bundles and assets from isolated application builds will be isolated from other builds.
+2. Isolated application builds can be deployed independently of *full builds* (non-isolated application builds).
+
+Updating the `build/generated` directory to follow this structure means that *full builds* will need to exclude the entry files of isolated applications, which ultimately excludes their bundles from being generated. The Webpack config will need to be updated to account for this, along with knowing when and where to store bundles and assets from isolated application builds.
+
+In the future, the subdirectories of isolated application bundles can be moved to `build/dist` to follow a more standard naming convention. 
+
+### Updating the content build for the new directory structure
+
+The `content-build` repository is responsible for creating the HTML files that are served on the site. By default, the build process in the `content-build` repository uses the `build/generated` path when injecting script tags that reference bundles and assets from `vets-website` deployments. This will need to be modified, since assets could be loaded from `build/generated` or one of the subdirectories for isolated application builds, e.g. `build/generated/vaos`. To better determine which paths in the deployment buckets to load bundles and assets from on each page, a generic script that handles this logic can be created. The build process in the `content-build` repository can then be updated to inject that generic script on each page. This essentially transfers control of determining which assets get loaded to `vets-website`, making it unlikely that production would break for applications migrating to the isolated structure.
```

Comment:
This would be used live on the site. So instead of the content-build hard coding the path of assets/bundles on pages, this script would initially load on pages and dynamically determine which application asset to reference on the page

Reply:
Have you all done any analysis of how this will affect performance?

Reply:
Current scripts are set to defer loading, so the performance impact should be minimal, if at all - the initial script will be very small so the load time should just be async loading of the app scripts, which we're already doing; We're following the model of Google analytics/typekit here, I think it will give us a lot more control in the longrun, but it would be good to try and offset any performance issues by improving something else on site (or at least just running an audit and generating that asset for later use).

The big question mark here is whether React will work when it's loaded async